File: 2025-1/sesion-de-clase-08 - Parte 02/SoftProg/SoftProgPersistencia/src/main/java/pe/edu/pucp/inf30/softprog/rrhh/dao/ICrud.java
package pe.edu.pucp.inf30.softprog.rrhh.dao;

import java.util.List;

/**
 *
 * @author eric
 * @param <T>
 */
public interface ICrud<T> {
    int insertar(T modelo);
    boolean modificar(T modelo);
    boolean eliminar(int id);
    T buscar(int id);
    List<T> listar();
}

File: 2025-1/sesion-de-clase-18/SoftProgNet/SoftProgPersistencia/DAO/ICrud.cs
using System.Collections.Generic;
using System.Data.Common;

namespace PUCP.Edu.Pe.SoftProg.Persistencia.DAO {
    public interface ICrud<T> {
        int Insertar(T modelo);
        int Insertar(T modelo, DbConnection conexion, DbTransaction transaction);
        bool Modificar(T modelo);
        bool Eliminar(int id);
        T Buscar(int id);
        List<T> Listar();
    }
}
